### PR TITLE
feature(event filtering): criteo onetag

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

- in browser.js file of criteo destination, in track call, logic added in order to map events

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [x] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/595)

### CODE REVIEW

File: integrations/Criteo/constants.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
+
 const NAME = "CRITEO";
+
 const CNameMapping = {
   [NAME]: NAME,
   Criteo: NAME,
   criteo: NAME,
 };
 
-export { NAME, CNameMapping };
```
Comment (ItsSudip) [inline integrations/Criteo/constants.js:8]: This file is not formatted.

File: integrations/utils/commonUtils.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import _ from "lodash";
-import logger from "../../utils/logUtil";
```
Comment (ItsSudip) [inline integrations/utils/commonUtils.js:2]: Is this not used anywhere in this file?

Comment (mihir-4116) [reply integrations/utils/commonUtils.js:2]: @ItsSudip , no we are not using it anywhere in file

**Review by ItsSudip — DISMISSED** [review]

File: integrations/Criteo/browser.js [hunk 3]
```diff
@@ -116,23 +118,47 @@ class Criteo {
...
+      return;
     }
+    let events = [];
+    if (supportedEvents.includes(trimmedEvent)) {
+      events.push(trimmedEvent);
+    } else {
+      events = eventMapping[trimmedEvent];
+    }
+
+    if (events.length > 0) {
```
Comment (saikumarrs) [inline integrations/Criteo/browser.js:139]: No need for this check. Array.forEach implicitly handles empty arrays.

File: integrations/utils/commonUtils.js [hunk 4]
```diff
@@ -40,9 +39,12 @@ const getHashFromArrayWithDuplicate = (
         : array[fromKey].trim();
 
       if (hashMap[key]) {
-        hashMap[key].push(array[toKey]);
+        const val = hashMap[key];
+        if (!val.includes(toKey)) {
```
Comment (saikumarrs) [inline integrations/utils/commonUtils.js:43]: What's the purpose of this check?

Comment (ItsSudip) [reply integrations/utils/commonUtils.js:43]: We are checking if duplicate elements are added to the array or not. Let's say someone provided same event name mapping twice. Something like:
```
[
          {
            "from": "Downloaded",
            "to": "download"
          },
          {
            "from": "Downloaded",
            "to": "download"
          }
        ]
```
then the Downloaded key will contain the array with ["download", "download"]. To omit this situation we are doing this check.

Comment (saikumarrs) [reply integrations/utils/commonUtils.js:43]: In that case, we should not be checking for the presence of `toKey` (which is the string literal `to`).

Comment (saikumarrs) [reply integrations/utils/commonUtils.js:43]: Shouldn't it be `if (!val.includes(array[toKey])) {`?

Comment (ItsSudip) [reply integrations/utils/commonUtils.js:43]: We decided to do that with help of set but that would be a breaking change.

Comment (saikumarrs) [reply integrations/utils/commonUtils.js:43]: I don't understand. It has a logical error.

Comment (ItsSudip) [reply integrations/utils/commonUtils.js]: Yes. Updated the code.

**Review by saikumarrs — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**